### PR TITLE
#1965: Handle js object in crash changelog output

### PR DESCRIPTION
**Pull request by frankhereford**
This pull request intends to close cityofaustin/atd-data-tech/issues/1965.

The changes include the detection of a JS object when printing a crash change log and using the JSON.stringify method on it to make it human readable.

### CODE REVIEW

**Comment by patrickm02L** [comment]
Deferring to the Devs for Review.

**Comment by mateoclarke** [comment]
@patrickm02L I just wanted to share with you the way I was able to do a visual review in addition to a code review...

Under the "All checks have passed" section, find the deploy preview for "netlify/atd-vze-staging/deploy-preview" and click "Details" to the right. This gives you the link to the [Deploy Preview](https://deploy-preview-1033--atd-vze-staging.netlify.app/) for the code in this PR.

From there, I compared a [crash in the staging instance](https://visionzero-staging.austinmobility.io/editor/#/crashes/18032221) of VZE which represents the `master` branch without this code against the [same crash in Frank's Deploy Preview](https://deploy-preview-1033--atd-vze-staging.netlify.app/#/crashes/18032221). 

At the bottom of the Crash details page in the Change Log section, I was able to see that where there was `[Object object]` in staging, there is now a stringified JSON with better human readable details for how `geocode_method` changed. See below.

### Before
Screen Shot 2021-10-06 at 1 26 24 PM

### After
Screen Shot 2021-10-06 at 1 26 19 PM

I repeated this for a few crashes just as a surface spot check.

**Review by mateoclarke — APPROVED** [review]
🚢 💨 ✨ 

michael-scott-wink

**Comment by patrickm02L** [comment]
@mateoclarke I did a visual review as well through the Netlify link and saw the change to {"name":"CRIS"}. 

This PR only called for one person to approve in order to become mergeable. I deferred to Devs to look at the code side of things because I didn't want to approve without making sure the code was straight.

File: atd-vze/src/views/Crashes/CrashChangeLog.js [hunk 1]
```diff
@@ -116,7 +116,30 @@ class CrashChangeLog extends Component {
...
+    let modalItems = diff.map(item => {
+
+      // Define a function to pass to JSON.stringify which will skip over
+      // the __typename key in the JS object being stringified.
+      const stringify_replacer = (key, value) => {
+        if (key === '__typename') {
+          return undefined;
+        }
+        return value;
+      }
```
[the remark this answers is not in the document]
Comment (frankhereford) [reply atd-vze/src/views/Crashes/CrashChangeLog.js:128]: Those are such good observations! I totally agree with both of them, and I would like to fix them both. I really appreciate you taking the time to write them up. I will keep them in mind going forward into other issues and projects.

**Pull request merged**